mlflow/tracing/types/wrapper.py

```diff
@@ -187,44 +272,110 @@ def end(self):
 
         self._span.end()
 
-    def to_mlflow_span(self) -> Span:
-        """
-        Create an MLflow Span object from this wrapper and the original Span object.
+    def to_dict(self):
+        # NB: OpenTelemetry Span has to_json() method, but it will write many fields that
+        #  we don't use e.g. links, kind, resource, trace_state, etc. So we manually
+        #  cherry-pick the fields we need here.
+        return {
+            "name": self.name,
+            "context": {
+                "span_id": self.span_id,
+                "trace_id": self._trace_id,
+            },
+            "parent_id": self.parent_id,
+            "start_time": self.start_time_ns,
+            "end_time": self.end_time_ns,
+            "status_code": self.status.status_code,
+            "status_message": self.status.description,
+            "attributes": dict(self._span.attributes),
```

Comment:
Feel like we should be consistent on the usage of self._span.attributes or self._attributes. Corresponding to my below comment in _SpanAttributesRegistry